### PR TITLE
added log4js for better debugging and /shelters/{shelterID}/pets

### PR DESCRIPTION
**Pull request by dotte-osu**

### CODE REVIEW

**Comment by dotte-osu** [comment]
- added log4js
- added /shelters/{shelterID}/pets

File: src/controllers/sheltersController.js [hunk 1]
```diff
@@ -48,23 +52,25 @@ const postShelters = async (req, res, next) => {
...
     .catch((e) => {
-      console.log(e);
+      log.error(e);
       res.sendStatus(500);
       next(e);
     });
 };
 
 const updateShelter = async (req, res, next) => {
+  log.debug("Calling getShelters...Verifying user inputs...");
```
Comment (asching7108) [inline src/controllers/sheltersController.js]: It should be updateShelters here

File: src/server.js [hunk 2]
```diff
@@ -27,6 +30,7 @@ router.post('/shelters', sheltersController.postShelters);
 router.get('/shelters/:shelterID', sheltersController.getShelter);
 router.delete('/shelters/:shelterID', sheltersController.deleteShelter);
 router.patch('/shelters/:shelterID', sheltersController.updateShelter);
+router.get('/shelters/:shelterID/pets', sheltersController.getPets);
```
Comment (asching7108) [inline src/server.js:32]: Just a side note, the getShelters route is still /shelters/test, do we want to remove the /test?

File: src/controllers/sheltersController.js [hunk 3]
```diff
@@ -87,18 +93,33 @@ const updateShelter = async (req, res, next) => {
...
+      res.sendStatus(500);
+      next(e);
+    });
+};
+
+const getPets = async (req, res, next) => {
+  log.debug("Calling getPets...");
+  await getAllPets(req.params.shelterID)
+    .then((dbResponse) => {
+      if (dbResponse.length == 0) res.sendStatus(404);
```
Comment (asching7108) [inline src/controllers/sheltersController.js]: I wonder if we would want to distinguish 'shelter not found' or 'shelter exists but has no pets yet' here. It depends on how we want to implement the frontend, but I imagine we may need a way to tell the difference? Like if the shelter id doesn't exist, we want to return an error or redirect the user to the not found page, but if the shelter simply doesn't have any pets yet we may want to just display an empty list and a create button. If we want to tell the difference we'll probably need to run a select query to the shelters table first in the model. Let me know what you think!

**Review by asching7108 — COMMENTED** [review]
Thank you for adding log4js!! It definitely will help a lot. I don't have a lot to add but mainly just a question about the getPets endpoint. All looks really good!

File: src/controllers/sheltersController.js (hunk 1 above)
Comment (dotte-osu) [reply src/controllers/sheltersController.js]: thank you for catching it!

File: src/server.js (hunk 2 above)
Comment (dotte-osu) [reply src/server.js:32]: Removed the test since we are not using it.

File: src/controllers/sheltersController.js (hunk 3 above)
Comment (dotte-osu) [reply src/controllers/sheltersController.js]: I agree, I changed the code to return below.
 'shelter not found': --> 404, "Shelter not found"
 'shelter exists but has no pets yet' --> 200. [] (empty array)

**Comment by dotte-osu** [comment]
Thank you for catching the small things. I fixed the debug comment, removed the sample route. I agree that we should distinguish 'shelter not found' or 'shelter exists but has no pets yet'. I changed the responses for it.

**Comment by dotte-osu** [comment]
Thank you for catching the small things. I fixed the debug comment, removed the sample route. I agree that we should distinguish 'shelter not found' or 'shelter exists but has no pets yet'. I changed the responses for it.

**Review by asching7108 — APPROVED** [review]

**Pull request merged**